pwndbg/lib/regs.py

```diff
@@ -19,7 +19,107 @@
 import pwndbg.lib.disasm.helpers as bit_math
 from pwndbg.lib.arch import PWNDBG_SUPPORTED_ARCHITECTURES_TYPE
 
-BitFlags = OrderedDict[str, Union[int, Tuple[int, int]]]
+
+class BitFlags:
+    # this should be backwards compatibility
+    reg: str  # this is intentionally uninitialized -- arm uses the same self.flags structuture for different registers
+    flags: OrderedDict[str, Union[int, Tuple[int, int]]]
+
+    def __init__(self, flags: List[Tuple[str, Union[int, Tuple[int, int]]]] = []):
+        self.reg = ""
+        self.flags = {}
+        for name, bits in flags:
+            self.flags[name] = bits
+
+    def __getattr__(self, name):
+        if name in {"reg", "value", "last"}:
+            return self.__dict__[name]
+        return getattr(self.flags, name)
+
+    def __getitem__(self, key):
+        return self.flags[key]
+
+    def __setitem__(self, key, value):
+        self.flags[key] = value
+
+    def __delitem__(self, key):
+        del self.flags[key]
+
+    def __iter__(self):
+        return iter(self.flags)
+
+    def __len__(self):
+        return len(self.flags)
+
+    def __repr__(self):
+        return f"BitFlags({self.flags})"
+
+    def update(self, regname: str):
+        self.reg = regname
+
+    def context(self, rc):
+        return rc.flag_register_context(self.reg, self)
+
+
+class AddressingRegister:
+    """
+    Represents a register that is used to store an address, e.g. cr3, gsbase, fsbase
+    """
+
+    reg: str
+    value: int
+    is_virtual: bool  # inicating if the address is a virtual address
```

Comment:
```suggestion
    is_virtual: bool
```
What does virtual mean here?